state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1247,18 +1282,22 @@ pub mod pallet {
 
 				let mut stable_amount = swap.stable_amount.unwrap_or_default();
 
-				let required_fees = &swap.swap.fees;
-				for fee_type in required_fees {
-					match fee_type {
+				for fee_type in &swap.swap.fees {
+					let remaining_amount = match fee_type {
 						FeeType::NetworkFee => {
-							let NetworkFeeTaken { remaining_amount, network_fee } =
+							let FeeTaken { remaining_amount, fee } =
 								Self::take_network_fee(stable_amount);
-							stable_amount = remaining_amount;
-							swap.network_fee_taken = Some(network_fee);
+							swap.network_fee_taken = Some(fee);
+							remaining_amount
 						},
-						// For now there is only one type of fee in this context, but we we expect
-						// broker fees to be processed here in the future too
-					}
+						FeeType::BrokerFee(beneficiaries) => {
+							let FeeTaken { remaining_amount, fee } =
+								Self::take_broker_fees(stable_amount, beneficiaries);
+							swap.broker_fee_taken = Some(fee);
+							remaining_amount
```

Comment:
~~I feel like we should also emit an event here containing the fee_taken info to enable the product to display this.~~

---
*edit: Just spoke with @zoheb391 and currently they reverse-engineer the information from SwapExecuted, ie. they look at broker_fee and split it up proportionally among the beneficiaries to figure out who got how much.*

*It would be better to accumulate the FeeTaken information in to a vec and emit this in the swap executed event. We can keep the 'legacy' network_fee and broker_fee fields.*